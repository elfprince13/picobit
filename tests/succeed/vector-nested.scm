;; (define vecA #"Hello World Vectors")
;; (print vecA)
;; (print (u8vector-length vecA))
;; (let ([bigvec (#%make-u8vector 1280 0)])
;;   (print bigvec))
;; 
;; (let ([bigvec (#%make-u8vector 6 0)])
;;   ; little-endian pair (bigvec . #f)
;;   (u8vector-set! bigvec 0 #x00)
;;   (u8vector-set! bigvec 1 #x85)
;;   (u8vector-set! bigvec 2 #x00)
;;   (u8vector-set! bigvec 3 #x86)
;;   (print bigvec)
;;   (let ([pair (cons bigvec #f)]
;;         [pair2 (cons #f #f)])
;;     (print pair)
;;     (print bigvec)))
;; 
;; (putchar #x0a)

(define vecB (#%make-vector 13 #"HWV"))
(print vecB)
(vector-set! vecB 3 (cons #t #f))
(print vecB)
(vector-set! vecB 4 (#%make-vector 9 6))
(print vecB)
(vector-set! vecB 5 "hello world")
(print vecB)
(putchar #x0a)