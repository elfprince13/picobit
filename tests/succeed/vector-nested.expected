'#('#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86))
'#('#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) (#t . #f) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86))
'#('#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) (#t . #f) '#(6 6 6 6 6 6 6 6 6) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86))
'#('#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) (#t . #f) '#(6 6 6 6 6 6 6 6 6) "hello world" '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86) '#u8(72 87 86))

